packages/core-typings/src/IMessage/IMessage.ts

```diff
@@ -249,6 +255,9 @@ export const isEditedMessage = (message: IMessage): message is IEditedMessage =>
 	'_id' in (message as IEditedMessage).editedBy &&
 	typeof (message as IEditedMessage).editedBy._id === 'string';
 
+export const isSystemMessage = (message: IMessage): message is ISystemMessage =>
+	't' in message && message.t !== undefined && MessageTypes.includes(message.t);
```

Comment:
```suggestion
	message.t !== undefined && MessageTypes.includes(message.t);
```

will work pretty much the same